scarb/src/ops/proc_macro_server/mod.rs

```diff
@@ -1,13 +1,41 @@
 use crate::compiler::plugin::proc_macro::ProcMacroHost;
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use connection::Connection;
+use json_rpc::ErrResponse;
+use proc_macro_server_api::RpcResponse;
 
 mod connection;
+mod json_rpc;
 
 pub fn start_proc_macro_server(proc_macros: ProcMacroHost) -> Result<()> {
     let connection = Connection::new();
 
-    //TODO
+    for i in 0..4 {
+        let receiver = connection.receiver.clone();
+        let sender = connection.sender.clone();
+
+        std::thread::Builder::new()
+            .name(format!("proc-macro-server-worker-thread-{i}"))
+            .spawn(move || {
+                for request in receiver {
+                    let response = match request.method.as_str() {
```

Comment:
the indentation is going crazy here, please:
1. extract worker body to a separate function
2. extract request method routing to another one